Test fails to compile with format error

Operating System: MacOS 15.4.1 (24E263

```
  CXX      test/test_bitcoin-serialize_tests.o
  CXX      test/test_bitcoin-sighash_tests.o
test/sighash_tests.cpp:34:65: error: format specifies type 'int' but the argument has type 'unsigned int' [-Werror,-Wformat]
   34 |         printf("ERROR: SignatureHash(): nIn=%d out of range\n", nIn);
      |                                             ~~                  ^~~
      |                                             %u
test/sighash_tests.cpp:61:70: error: format specifies type 'int' but the argument has type 'unsigned int' [-Werror,-Wformat]
   61 |             printf("ERROR: SignatureHash(): nOut=%d out of range\n", nOut);
      |                                                  ~~                  ^~~~
      |                                                  %u
2 errors generated.
make[3]: *** [test/test_bitcoin-sighash_tests.o] Error 1
make[2]: *** [all-recursive] Error 1
make[1]: *** [all] Error 2

```

--- a/src/test/sighash_tests.cpp
+++ b/src/test/sighash_tests.cpp
@@ -31,7 +31,7 @@ uint256 static SignatureHashOld(CScript scriptCode, const CTransaction& txTo, un
     static const uint256 one(uint256S("0000000000000000000000000000000000000000000000000000000000000001"));
     if (nIn >= txTo.vin.size())
     {
-        printf("ERROR: SignatureHash(): nIn=%d out of range\n", nIn);
+        printf("ERROR: SignatureHash(): nIn=%u out of range\n", nIn);
         return one;
     }
     CMutableTransaction txTmp(txTo);
@@ -58,7 +58,7 @@ uint256 static SignatureHashOld(CScript scriptCode, const CTransaction& txTo, un
         unsigned int nOut = nIn;
         if (nOut >= txTmp.vout.size())
         {
-            printf("ERROR: SignatureHash(): nOut=%d out of range\n", nOut);
+            printf("ERROR: SignatureHash(): nOut=%u out of range\n", nOut);
             return one;
         }
         txTmp.vout.resize(nOut+1);

